bank_statement_check_number/models/account_bank_statement.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = "account.bank.statement.line"
+
+    check_number = fields.Char(
+        copy=False,
+        index=True,
+    )
+
+    @api.model
+    def _prepare_move_line_default_vals(self, counterpart_account_id=None):
+        move_lines = super()._prepare_move_line_default_vals(counterpart_account_id)
+        for move_line in move_lines:
+            move_line.update({"check_number": self.check_number})
+        return move_lines
```

Comment:
```suggestion
    @api.model
    def _prepare_move_line_default_vals(self, counterpart_account_id=None):
        move_lines = super()._prepare_move_line_default_vals(counterpart_account_id)
        for move_line in move_lines:
            move_line.update({"check_number": self.check_number})
        return move_lines
```

- consider removing `@api.model` if the method works on a recordset (`self.ensure_one()` is used).
- when overriding a method, make sure to keep the same arguments as the original definition for compatibility.
- rename `move_lines` to something more descriptive like `line_vals_list` to reflect that it's a list of value dictionaries, not actual move line records.